proposals/yield-in-try-catch.md

```diff
@@ -172,100 +170,130 @@ void IDisposable.Dispose()
 
 This behavior is important to understand when considering the code generation for `try / catch` blocks.
 
-### Code generation yield inside try with catch
-
-The code generation for iterators that have `yield` inside `try` blocks must preserve the same exception semantics for the `catch` in both `MoveNext` and `Dispose`. To achieve this the compiler will take a similar approach to what it does for `finally` blocks.
-
-For every `catch` block where the `try` has a nested `try / finally` with `yield`:
-
-1. The contents of the `catch` block will be generated into a parameterless method on the iterator with a `void` return.
-2. The contents of the `when` clause will be generated into a parameterless method on the iterator with a `bool` return.
-3. The `catch` block will be replaced with a call to the generated method in `MoveNext`.
-4. The `when` clause will be replaced with a call to the generated method in `MoveNext`.
-5. The `Dispose` method will mirror `catch` and `finally` blocks in the same way it mirrors `finally` blocks today and dispatch to the appropriate method
-
-The exception object will be lifted into a field just as any other local would be and accesses to it in the generated `catch` and `when` methods will use the field. In the case the language allowed `when` clauses the pattern variables would be lifted into fields as well.
+### Code generation of yield inside try with catch in iterators
 
-For example consider this code sample:
+The code generation for iterators that have `yield` inside a `try / catch` will preserve the `catch` semantics in the `Dispose` method. That is _only_ the contents of the `finally` blocks will execute even in the case exceptions come into play. For example:
 
 ```csharp
-static IEnumerable<int> Iterator()
+var e = M1(true).GetEnumerator();
+e.MoveNext();
+e.Dispose();
+
+IEnumerable<int> M1(bool b)
 {
     try
     {
         try
         {
-            yield return M(1);
+            try
+            {
+                yield return 13;
+            }
+            finally
+            {
+                Console.WriteLine("inner finally");
+                if (b)
+                {
+                    throw new Exception();
+                }
+            }
         }
-        finally
+        catch
         {
-            Console.WriteLine("Finally Inner");
+            Console.WriteLine("catch");
         }
-    }
-    catch (InvalidOperationException ex1) when (ex.Message.Contains("hello"))
-    {
-        Console.WriteLine("Catch1");
-    }
-    catch (Exception ex2) 
-    {
-        Console.WriteLine("Catch2");
+
+        Console.WriteLine("after catch");
     }
     finally
     {
-        Console.WriteLine("Finally Outer");
+        Console.WriteLine("outer finally");
     }
 }
-static T M<T>(T t) => t;
 ```
 
-Will generate the following `Dispose` method:
+This program will output:
+
+```cmd
+inner finally
+outer finally
```

Comment:
Consider adding the exception to the output since it seems it will be thrown from the Dispose invocation, right?

Reply:
Yse good point.